### DISCUSSION TITLE
Help: erratic "user font not loaded" issue

### DISCUSSION THREAD

**Discussion by mdmcconnell** [pdfcpu/pdfcpu#889] [Q&A] [answered]
I've written a small program that fills pdf form information using api.FillFormFile.  Using the same input files and all other conditions, running the program functions as expected some of the time.  Other runs, it exits with the message "pdfcpu: user font not loaded: HelveticaLTStd-Bold".  I imagine it could be something exogenous in the file system or on my machine, causing this font not to load.  The likelihood of hitting this error seems to increase with the number of form fields I attempt to process (ie, if I stop after five, ~50% chance of success; if I go for 50, 0% observed).  I'm stuck.  What does this error mean?  Where do user fonts come from? I'm not specifying any formatting in my program.  Can it be from the source pdf file?  Is there a way to check or force that all required fonts are loaded before attempting to write?  Or ignore font requests entirely?  Finally, why does the api include code that calls exit() instead of giving me some way to recover?

**Comment by mdmcconnell** [comment]
Update - I'm now replicating this issue with the CLI, using form export and form fill, so it doesn't appear to be my code only.

**Comment by mdmcconnell** [comment]
Here are steps to replicate, using the linked form:
```
$ pdfcpu form export f1040.pdf ex.json
$ jq '.forms[].textfield[] |= (.value = .id)' ex.json > fill.json
$ pdfcpu form fill f1040.pdf fill.json out.pdf
```
Result:
```
writing out.pdf...
optimizing...
removing signature...
filling...
pdfcpu: user font not loaded: HelveticaLTStd-Bold
```
[f1040.pdf](https://github.com/user-attachments/files/15842641/f1040.pdf)

**Comment by mdmcconnell** [comment, marked as answer]
This is duplicate of https://github.com/pdfcpu/pdfcpu/issues/767 and fixed in dev code.  Sorry for the trouble.

**Comment by hhrutter** [comment]
You're welcome!
Thanks for using pdfcpu 💚
